tools/codegen/model.py

```diff
@@ -118,13 +118,23 @@ def __str__(self) -> str:
     def lower(self) -> str:
         return str(self).lower()
 
+    def is_autograd_key(self) -> bool:
+        return 'Autograd' in str(self)
+
     @staticmethod
     def parse(value: str) -> 'DispatchKey':
         for k, v in DispatchKey.__members__.items():
             if k == value:
                 return v
         raise AssertionError(f'unknown dispatch key {value}')
 
+    @staticmethod
+    def try_parse(value: str) -> Optional['DispatchKey']:
+        try:
+            return DispatchKey.parse(value)
+        except AssertionError:
+            return None
```

Comment:
Will be wondering why we need this. If we actually need to detect parse errors we'll need to roll out a separate exception hierarchy for them; catching assert errors is too dangerous

Reply:
Fair; I can rewrite this to avoid the try/catch. Alternatively, if we're fine with the original parsing error message then we don't need this